src/nodes/composite/predicate.ts

```diff
@@ -403,7 +442,9 @@ type functionalConstraints<input> = {
     morph?: listable<Morph<input>>
 }
 
-type classConstraints<base extends Constructor> = base extends typeof Array
+type classConstraints<base extends Constructor> = base extends
+    | typeof Array
+    | typeof Date
     ? {
           props?: PropsInput
           range?: Range
```

Comment:
We should have the type of `Range` be generic with a parameter of the bounded type (or could just be something like kind === "numeric" | "date") so that depending on `base`, limit would have to be either a number or Date.

You can then add a `nodeDefinitions` test to make sure you get an error on a mismatch like we talked about for the other stuff we ended up not needing.